data/data/alibabacloud/vpc/sg-worker.tf

```diff
@@ -0,0 +1,272 @@
+  ip_protocol              = "tcp"
+  policy                   = "accept"
+  port_range               = "9000/9999"
+  security_group_id        = alicloud_security_group.sg_worker.id
+  source_security_group_id = alicloud_security_group.sg_master.id
+}
+
+resource "alicloud_security_group_rule" "sg_rule_worker_ingress_internal_udp" {
+  type              = "ingress"
+  ip_protocol       = "udp"
+  policy            = "accept"
+  port_range        = "9000/9999"
+  security_group_id = alicloud_security_group.sg_worker.id
+  cidr_ip           = var.vpc_cidr_block
+}
+
+resource "alicloud_security_group_rule" "sg_rule_worker_ingress_internal_from_master_udp" {
+  type                     = "ingress"
+  ip_protocol              = "udp"
+  policy                   = "accept"
+  port_range               = "9000/9999"
+  security_group_id        = alicloud_security_group.sg_worker.id
+  source_security_group_id = alicloud_security_group.sg_master.id
+}
+
+resource "alicloud_security_group_rule" "sg_rule_worker_ingress_kube_scheduler_from_master" {
+  type              = "ingress"
+  ip_protocol       = "tcp"
+  policy            = "accept"
+  port_range        = "10259/10259"
+  security_group_id = alicloud_security_group.sg_worker.id
+  cidr_ip           = var.vpc_cidr_block
+}
+
+resource "alicloud_security_group_rule" "sg_rule_worker_kube_controller_manager" {
+  type              = "ingress"
+  ip_protocol       = "tcp"
+  policy            = "accept"
+  port_range        = "10257/10257"
+  security_group_id = alicloud_security_group.sg_worker.id
+  cidr_ip           = var.vpc_cidr_block
+}
+
+resource "alicloud_security_group_rule" "sg_rule_worker_ingress_kube_controller_manager_from_master" {
+  type                     = "ingress"
+  ip_protocol              = "tcp"
+  policy                   = "accept"
+  port_range               = "10257/10257"
+  security_group_id        = alicloud_security_group.sg_worker.id
+  source_security_group_id = alicloud_security_group.sg_master.id
+}
+
+resource "alicloud_security_group_rule" "sg_rule_worker_ingress_kubelet_secure" {
+  type                     = "ingress"
+  ip_protocol              = "tcp"
+  policy                   = "accept"
+  port_range               = "10250/10250"
+  security_group_id        = alicloud_security_group.sg_worker.id
+  source_security_group_id = alicloud_security_group.sg_master.id
+}
+
+resource "alicloud_security_group_rule" "sg_rule_worker_ingress_kubelet_secure_from_master" {
+  type                     = "ingress"
+  ip_protocol              = "tcp"
+  policy                   = "accept"
+  port_range               = "10250/10250"
+  security_group_id        = alicloud_security_group.sg_worker.id
+  source_security_group_id = alicloud_security_group.sg_master.id
+}
+
+resource "alicloud_security_group_rule" "sg_rule_worker_ingress_etcd" {
+  type              = "ingress"
+  ip_protocol       = "tcp"
+  policy            = "accept"
+  port_range        = "2379/2380"
```

Comment:
I do not believe the worker security group should will access etcd on these ports. The etcd service will only be accessed by the master's themselves.

Reply:
Finished